### PR TITLE
Simplified main.py + split overrides

### PR DESCRIPTION
**Pull request by jacknagz**
This addresses feedback about simplifying the boilerplate `main.py` and separating overrides into a separate file. Overall, this pattern is much cleaner. 

One area to pay close attention to is setting individual overrides after running `get_rules` or `get_panther_rules.` The imports are much more definitive and reliable than referencing fetched rules by ID, but has the downside of not knowing whether or not they're included in the fetched rules. If you have a suggestion/opinion about this, please let me know in the comments!

### Changes
- Move overrides into two overrides files and import them into `main.py`
- Update the `validate` rule to be more realistic
- Remove the `filters` folder - it's cleaner to be in `overrides`
- Simplify the `main.py` significantly and remove unneeded comments/code

### CODE REVIEW

**Comment by debugmiller** [comment]
I like this change. But I worry that the `main.py` has too many examples. *every* customer that wants to get started has to read through and understand each line of the `main.py` and its not clear what is optional vs required vs recommended. I would suggest that any optional portion be commented out so that customers can quickly understand the difference.

File: main.py [hunk 1]
```diff
@@ -1,147 +1,32 @@
...
+        LogType.AWS_GUARDDUTY,
+        LogType.OKTA_SYSTEM_LOG,
+        LogType.GITHUB_AUDIT,
+        LogType.PANTHER_AUDIT,
+    ],
+    default_severity=[
+        Severity.MEDIUM,
+        Severity.HIGH,
+        Severity.CRITICAL,
+    ],
```
Comment (debugmiller) [inline main.py:20]: for example these kwargs could be commented out to make it clear it is optional

File: main.py [hunk 2]
```diff
@@ -1,147 +1,32 @@
...
 )
+# Load all local custom rules
+custom_rules = get_rules(module=rules)
+# Set a required field through a direct import
+PantherAuditUploadArtifacts.allowed_users = ["PAT Upload"]
```
Comment (debugmiller) [inline main.py:25]: same here

File: main.py [hunk 3]
```diff
@@ -1,147 +1,32 @@
...
-register(
-    [
-        AWSCloudTrailStopped,
-        AWSCloudTrailAccountDiscovery,
-        AWSConsoleRootLogin,
-    ]
-)
+# Apply overrides
+aws_cloudtrail.apply_overrides(base_rules)
+aws_guardduty.apply_overrides(base_rules)
```
Comment (debugmiller) [inline main.py:30]: and here

File: main.py [hunk 4]
```diff
@@ -1,147 +1,32 @@
...
-    default_runbook=(
-        "If the account is in production, investigate why CloudTrail was stopped. "
-        "If it was intentional, ensure that the account is monitored by another CloudTrail. "
-        "If it was not intentional, investigate the account for unauthorized access."
-    ),
-    reports=AWSCloudTrailStopped.reports | {"Internal": ["C.4"]},
+from overrides import aws_cloudtrail, aws_guardduty
+from rules.panther_audit.validate_rule import PantherAuditUploadArtifacts
+
+# Setup Manager and load rules
```
Comment (maxrichie5) [inline main.py:7]:
```suggestion
```

**Review by maxrichie5 — APPROVED** [review]
This looks really good!

**Review by dekatzenel — APPROVED** [review]

**Pull request merged**